service/resource/legacy/resource.go

```diff
@@ -0,0 +1,372 @@
+	return nil, nil
+}
+
+func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interface{}, error) {
+	return nil, nil
+}
+
+func (r *Resource) GetCreateState(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, error) {
+	var spec flanneltpr.Spec
+	{
+		o, ok := obj.(*flanneltpr.CustomObject)
+		if !ok {
+			return nil, microerror.Maskf(wrongTypeError, "expected '%T', got '%T'", &flanneltpr.CustomObject{}, obj)
+		}
+		spec = o.Spec
+	}
+
+	// Create flannel etcd config.
+	{
+		path := etcdNetworkConfigPath(spec)
+
+		type flannelBackend struct {
+			Type string
+			VNI  int
+		}
+
+		type flannelConfig struct {
+			Network   string
+			SubnetLen int
+			Backend   flannelBackend
+		}
+
+		config := flannelConfig{
+			Network:   spec.Flannel.Spec.Network,
+			SubnetLen: spec.Flannel.Spec.SubnetLen,
+			Backend: flannelBackend{
+				Type: "vxlan",
+				VNI:  spec.Flannel.Spec.VNI,
+			},
+		}
+
+		bytes, err := json.Marshal(config)
+		if err != nil {
+			return nil, microerror.Maskf(err, "marshaling %#v", config)
+		}
+
+		exists, err := r.store.Exists(context.TODO(), path)
+		if err != nil {
+			return nil, microerror.Maskf(err, "checking %s etcd key existence", path)
+		}
+		if exists {
+			r.logger.Log("debug", "etcd key "+path+" already exists", "event", "add", "cluster", spec.Cluster.ID)
+		} else {
+			err := r.store.Create(context.TODO(), path, string(bytes))
+			if err != nil {
+				return nil, microerror.Maskf(err, "createing %s etcd key", path)
+			}
+		}
+	}
+
+	// Create namespace for the cleanup job.
+	{
+		ns := newNamespace(spec, networkNamespace(spec))
+		_, err := r.k8sClient.CoreV1().Namespaces().Create(ns)
+		if apierrors.IsAlreadyExists(err) {
+			r.logger.Log("debug", "namespace "+ns.Name+" already exists", "event", "add", "cluster", spec.Cluster.ID)
+		} else if err != nil {
+			return nil, microerror.Maskf(err, "creating namespace %s", ns.Name)
+		}
+	}
+
+	// Create a dameonset running flanneld and creating network bridge.
+	{
+		daemonSet := newDaemonSet(spec, r.etcdCAFile, r.etcdCrtFile, r.etcdKeyFile)
+		_, err := r.k8sClient.ExtensionsV1beta1().DaemonSets(networkNamespace(spec)).Create(daemonSet)
+		if apierrors.IsAlreadyExists(err) {
+			r.logger.Log("debug", "daemonSet "+daemonSet.Name+" already exists", "event", "add", "cluster", spec.Cluster.ID)
+		} else if err != nil {
+			return nil, microerror.Maskf(err, "creating daemonSet %s", daemonSet.Name)
+		}
+	}
+
+	r.logger.Log("info", "started flanneld", "event", "add", "cluster", spec.Cluster.ID)
+
+	return nil, nil
+}
+
+func (r *Resource) GetDeleteState(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, error) {
+	var spec flanneltpr.Spec
+	{
+		o, ok := obj.(*flanneltpr.CustomObject)
+		if !ok {
+			return nil, microerror.Maskf(wrongTypeError, "expected '%T', got '%T'", &flanneltpr.CustomObject{}, obj)
+		}
+		spec = o.Spec
+	}
```

Comment:
We have to execute the old logic here because the old code does not separate between algorithm, data and dependency invocation. We just need an entrypoint and `GetDeleteState` is the only safe entrypoint for deletion events because `ProcessDeleteState` is also executed on create and update events.